process/smartContract/sovereignSCProcess.go

```diff
@@ -0,0 +1,649 @@
+package smartContract
+
+import (
+	"bytes"
+	"fmt"
+	"math/big"
+	"sort"
+	"strings"
+
+	"github.com/multiversx/mx-chain-core-go/core"
+	"github.com/multiversx/mx-chain-core-go/core/check"
+	"github.com/multiversx/mx-chain-core-go/data"
+	"github.com/multiversx/mx-chain-core-go/data/smartContractResult"
+	vmData "github.com/multiversx/mx-chain-core-go/data/vm"
+	"github.com/multiversx/mx-chain-core-go/hashing"
+	"github.com/multiversx/mx-chain-core-go/marshal"
+	"github.com/multiversx/mx-chain-go/common"
+	"github.com/multiversx/mx-chain-go/errors"
+	"github.com/multiversx/mx-chain-go/process"
+	"github.com/multiversx/mx-chain-go/sharding"
+	"github.com/multiversx/mx-chain-go/state"
+	"github.com/multiversx/mx-chain-go/storage"
+	logger "github.com/multiversx/mx-chain-logger-go"
+	vmcommon "github.com/multiversx/mx-chain-vm-common-go"
+)
+
+type sovereignSCProcessor struct {
```

Comment:
> you could have type sovereignSCProcessor struct { *smartContractProcessor } and you need to redefine only a few functions. less copy pasted code.

Right, changed it so that the constructor takes the scrProc as param

>plus we must make a merge between feat/vm1.5 to feat/chain-sdk - as you should directly work in scProcessorV2

Good point. Added a TODO for this

Reply:
Changes weren't pushed yet, but it is live now